src/Foundation/NSDate.cs

```diff
@@ -42,10 +42,19 @@ public partial class NSDate {
 		public static explicit operator DateTime (NSDate d)
 		{
 			double secs = d.SecondsSinceReferenceDate;
-			if ((secs < -63113904000) || (secs > 252423993599))
-				throw new ArgumentOutOfRangeException ("Value is outside the range of NSDate");
 
-			return new DateTime ((long)(secs * TimeSpan.TicksPerSecond + NSDATE_TICKS), DateTimeKind.Utc);
+			if ((secs < -63114076800) || (secs > 252423993599))
+				throw new ArgumentOutOfRangeException ($"Value is outside the range of NSDate {secs}");
+
+			var calendar = new NSCalendar (NSCalendarType.Gregorian);
```

Comment:
Memory leak, wrap the suave of the calendar in a using clause.

Reply:
not a _real_ leak, but better dispose right away (as this can be called a lot of time) than letting the GC handle it